uaclient/log.py

```diff
@@ -0,0 +1,17 @@
+import logging
+
+from uaclient import util
+
+
+class RedactionFilter(logging.Filter):
+    """A logging filter to redact confidential info"""
+
+    def redact_log(self, content: str) -> str:
+        if not isinstance(content, str):
+            content = str(content)
+        redacted_log = util.redact_sensitive_logs(content)
+        return redacted_log
+
+    def filter(self, record: logging.LogRecord):
+        record.msg = self.redact_log(record.msg)
+        return True
```

Comment:
Can these be combined into one function?

```suggestion
    def filter(self, record: logging.LogRecord):
        record.msg = util.redact_sensitive_logs(str(record.msg))
        return True
```

Reply:
Oh yes, Ithought of this approach but then I noticed it fails when we pass the exception object to it. Let me confirm that it does not fail or if I can find a better way of logging it.

Reply:
Oh yes, thanks.